firedrake/slate/slac/compiler.py

```diff
@@ -462,14 +466,17 @@ def facet_integral_loop(cxt_kernel, builder, coordsym, cellfacetsym,
     incl = []
     funcalls = []
     checker = chker[it_type]
+    split_forms = cxt_kernel.form_indices_map
     for splitkernel in cxt_kernel.tsfc_kernels:
         index = splitkernel.indices
         kinfo = splitkernel.kinfo
 
         # Generate an iterable of coefficients to pass to the subkernel
-        # if any are required
-        clist = [c for ci in kinfo.coefficient_map
-                 for c in builder.coefficient(exp.coefficients()[ci])]
+        # if any are required.
+        # Split form for this particular splitkernel
+        sform = split_forms[index]
+        clist = [c for i in kinfo.coefficient_map
+                 for c in builder.coefficient(sform.coefficients()[i])]
```

Comment:
OK, so rather than indexing into the original tensor expression's coefficients, you instead magically remember the split form, and ask it for its coefficients.  I think a much better thing to do is for this mapping to live in the `ContextKernel`, then you can map from the coefficients of the `tensor` directly to the relevant sub-bit.